esphome/components/web_server_idf/web_server_idf.cpp

```diff
@@ -348,7 +350,14 @@ void AsyncWebServerResponse::addHeader(const char *name, const char *value) {
   httpd_resp_set_hdr(*this->req_, name, value);
 }
 
-void AsyncResponseStream::print(float value) { this->print(to_string(value)); }
+void AsyncResponseStream::print(float value) {
+  // Use stack buffer to avoid temporary string allocation
+  // Size: sign (1) + digits (10) + decimal (1) + precision (6) + exponent (5) + null (1) = 24, use 32 for safety
+  constexpr size_t float_buf_size = 32;
+  char buf[float_buf_size];
+  int len = snprintf(buf, float_buf_size, "%f", value);
```

Comment:
Constant value still only specified once, same generated code, one line less source, one less variable name to comprehend, and `buf, sizeof(buf)` is self-evidently correct without reference to any other code.
```suggestion
  char buf[32];
  int len = snprintf(buf, sizeof(buf), "%f", value);
```